### PR TITLE
Truncate Long SQL Query

### LINKED ISSUE #74: Single segment payloads larger than 64KB throw warning
From https://github.com/humanmade/aws-xray/issues/14#issuecomment-642500187 this is an expected case.

Although can we maybe attempt to avoid the warning, by not sending the segment if it is larger than the UDP packet size?

In our case, we have a few long database queries which keep filling up in the error logs. These segments are anyway not delivered to the demon so I am wondering we might skip from sending those by checking the length?

### PR DESCRIPTION
**Pull request by ivankristianto**
Fixed #74 

This PR is to ensure we truncate the long query to max 5KB. As this is enough for any accident analysis on XRay reporting.

### CODE REVIEW

File: inc/namespace.php [hunk 1]
```diff
@@ -801,3 +804,21 @@ function redact_metadata( $metadata ) {
 
 	return $redacted;
 }
+
+/**
+ * Truncate sql query to given max_size.
+ *
+ * @param string $query       SQL Query to truncate.
+ * @param int    $max_size    Maximum size, default 5KB.
```
Comment (joehoyle) [inline inc/namespace.php:812]:
```suggestion
 * @param int    $max_size    Maximum size in bytes, default 5KB.
```

File: inc/namespace.php [hunk 2]
```diff
@@ -801,3 +804,21 @@ function redact_metadata( $metadata ) {
 
 	return $redacted;
 }
+
+/**
+ * Truncate sql query to given max_size.
```
Comment (joehoyle) [inline inc/namespace.php:809]:
```suggestion
 * Truncate sql query to given maximum size.
```

File: inc/namespace.php [hunk 3]
```diff
@@ -801,3 +804,21 @@ function redact_metadata( $metadata ) {
...
+ *
+ * @param string $query       SQL Query to truncate.
+ * @param int    $max_size    Maximum size, default 5KB.
+ * @param string $replacement String replacement.
+ *
+ * @return string
+ */
+function truncate_query( string $query, int $max_size = 5 * 1024, string $replacement = '…' ): string {
```
Comment (joehoyle) [inline inc/namespace.php:817]:
```suggestion
function truncate_query( string $query, int $max_size = 5 * 1024, string $replacement = '…' ) : string {
```

File: inc/namespace.php [hunk 4]
```diff
@@ -185,6 +185,9 @@ function trace_requests_request( $url, $headers, $data, $method, $options ) {
 }
 
 function trace_wpdb_query( string $query, float $start_time, float $end_time, $errored, $host = null ) {
+	// Truncate long query to ensure not exceed max limit.
```
Comment (joehoyle) [inline inc/namespace.php:188]:
```suggestion
	// Truncate long query to ensure it does not exceed max limit.
```

**Review by joehoyle — CHANGES_REQUESTED** [review]

**Comment by joehoyle** [comment]
@ivankristianto is this ready for re-review?

**Comment by ivankristianto** [comment]
ahh yes, @joehoyle 
Sorry I forgot to mention you. This PR ready for re-review.

**Review by joehoyle — APPROVED** [review]

**Pull request merged**